scripts/publish-documentation/publish.sh

```diff
@@ -116,22 +119,38 @@ do
   git switch gh-pages
   git clean -fdx
 
+  # For some reason, recent usage of this script failed since `.config` folder
+  # was preventing from making a commit, as it was "untracked" change.
+  # Therefore, this folder is pre-emptively removed here, just in case.
+  # We don't need it in `gh-pages` branch anyway.
+  log "[$PWD] Removing ./config folder."
+  rm -rf "./config"
+
   for path in $(echo "$paths" | tr "," "\n")
   do
     # Extracts the module name from a path. A path should use the "/" as a file separator.
     module="${path##*\/}"
 
+    log "[$PWD] Creating directory 'dokka-reference/$module/v/$version'"
     mkdir -p "dokka-reference/$module/v/$version"
-    cp -r "../$module/build/docs/dokka/" "dokka-reference/$module/v/$version/"
+
+    # Previous versions of Dokka produced their output in `dokka` subfolder.
+    # Most recent versions do that differently, generating the documentation
+    # into `dokkaJava` folder when used in JVM mode.
+    # We treat the observed behaviour as "new normal".
+    log "[$PWD] Copying recursively: '../$module/build/docs/dokkaJava/' -> 'dokka-reference/$module/v/$version/'"
+    cp -r "../$module/build/docs/dokkaJava/" "dokka-reference/$module/v/$version/"
 
     commit_message="Publish Dokka documentation for \`$module\` of \`$version\` "
     if [ $is_primary = true ]; then
-      cp -r "../$module/build/docs/dokka/" "dokka-reference/$module/"
+      log "[$PWD] PRIMARY Copying recursively: '../$module/build/docs/dokkaJava/' -> 'dokka-reference/$module/'"
+      cp -r "../$module/build/docs/dokkaJava/" "dokka-reference/$module/"
       commit_message+="as primary"
     else
       commit_message+="as secondary"
     fi
 
+    log "[$PWD] Executing 'git add \"dokka-reference/$module/*\"', and committing."
```

Comment:
`Staging and committing all changes in the \"dokka-reference/$module/*\" `seems shorter and conveys the same meaning.